### PR TITLE
DE: Committee Scraper Updated and MA:Vote Scraper Updated and HI: "carried over" bills bug solved

### PR DESCRIPTION
**Pull request by hiteshgarg14**
#1445 
Scraper is now working for `senate`, `house` and `joint` sections.

### CODE REVIEW

**Comment by hiteshgarg14** [comment]
#1162 
Vote Scraper Updated.

**Comment by hiteshgarg14** [comment]
#1196 
HI: "carried over" bills bug solved.

File: openstates/de/committees.py [hunk 1]
```diff
@@ -8,104 +8,87 @@ class DECommitteeScraper(CommitteeScraper,LXMLMixin):
...
+                if chair_man not in member and vice_chair not in member:
+                    committee.add_member(member)
+
+            committee.add_source(comm_url)
+            committee.add_source(url)
+
+            self.save_committee(committee)
+
 
     def scrape_joint_committees(self,term,session):
```
Comment (jamesturk) [inline openstates/de/committees.py]: is this just a copy pasted version of the other committee scraper?  if so let's go ahead & combine these

File: openstates/de/committees.py [hunk 2]
```diff
@@ -8,104 +8,87 @@ class DECommitteeScraper(CommitteeScraper,LXMLMixin):
...
+        for item in data:
+            committee = Committee(chamber, item['CommitteeName'])
+            chair_man = str(item.get('ChairName', 'No'))
+            vice_chair = str(item.get('ViceChairName', 'No'))
+            comm_id = item['CommitteeId']
+            comm_url = 'http://legis.delaware.gov/CommitteeDetail?' + \
+                        'committeeId='+ str(comm_id)
+            comm_page = lxml.html.fromstring(self.get(comm_url).text)
+            #all members including chair_man and vice_chair
+            members = comm_page.xpath("//section[@class='section-short']/" + \
```
Comment (jamesturk) [inline openstates/de/committees.py]: these line continuations aren't needed since you're inside parens

File: openstates/ma/votes.py [hunk 3]
```diff
@@ -1,209 +1,131 @@
...
+        final = []
+        yes = []
+        no = []
+        for i in wordList:
+        	for j in i.split("\n"):
+                 final.append(j)
+
+        c = 0
+        for i in final:
+        	if c == 1 and i != '  NAYS.' and i != '':
```
Comment (jamesturk) [inline openstates/ma/votes.py]: please use 4-spaces for indentation

File: openstates/de/committees.py (hunk 1 above)
Comment (hiteshgarg14) [inline openstates/de/committees.py]: Actually, in URLs variable, I added 'Joint' 
```
urls = {
            'upper': 'http://legis.delaware.gov/json/Committees/' +
                     'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=1',
            'lower': 'http://legis.delaware.gov/json/Committees/' +
                     'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=2',
           'joint': 'http://legis.delaware.gov/json/Committees/' + 
              'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=3'
        }
```

`url = urls[chamber] % (session,)`

 But on running scraper, only `upper` and `lower` were values of `chamber` variable. I tried it by adding `joint` in  __init__.py file's `chamber` dictionary but nothing happened. that's why I used `scrape_joint_committees` method. This is the same technique which was used before to scrape the `joint` committees.

**Comment by hiteshgarg14** [comment]
Hello :) I have made changes and fixed the committee scraper bug.

When in JSON data if we get `ChairName` field as `null` then it will be saved as `No`. similarly for `ViceChairName` field and in members name, I have fixed the bug. I was unable to check it locally as I have converted DE legislators.py to pupa. please have a look and sorry for one PR instead of 3 PR's.

**Comment by jamesturk** [comment]
there's now partial pupa work on this branch, you will  need to separate this out into multiple branches so that we can pull in parts of it without the pupa bits.  it might be best to separate it into multiple PRs at that point too

**Comment by hiteshgarg14** [comment]
Opened 3 separate PR's with bugs fixed. Please have a look. 
Closing this PR.

**Pull request closed without merging**